Fix cats getting instakilled by mousetraps

PGrayCS opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
Cats are now immune to mousetraps and display a popup when avoiding them.

## Why we need to add this
Fixes #2279 - Cats were getting instakilled by mousetraps, which doesn't make sense since cats are natural predators of mice and would instinctively avoid such traps. This change makes cats immune to mousetraps and adds flavor text when they avoid one.

## Media (Video/Screenshots)
N/A - Simple gameplay logic change. When a cat walks over a mousetrap, instead of dying, players nearby will see a popup: \The cat deftly avoids the mousetrap!\

## Checks

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
:cl: PGrayCS
- fix: Fixed cats getting instakilled by mousetraps.
- add: Added popup text when cats avoid mousetraps.

PGrayCS committed:
Fix cats getting instakilled by mousetraps

Fixes #2279

- Added ProtectedFromStepTriggers component to MobCat so all cats are immune to mousetraps
- Removed redundant ProtectedFromStepTriggers from MobCatSpace (inherits from MobCat)
- Added MousetrapAvoidedEvent to show emote text when cat avoids a mousetrap
- Created server-side MousetrapSystem to handle the chat message
- Added localization for the cat avoid emote

CrazyPhantom779 commented on Content.Shared/StepTrigger/Systems/StepTriggerImmuneSystem.cs:
```diff
@@ -1,5 +1,6 @@
 ﻿using Content.Shared.Examine;
 using Content.Shared.Inventory;
+using Content.Shared.Mousetrap;
 using Content.Shared.StepTrigger.Components;
```
```suggestion
using Content.Shared.StepTrigger.Components;
using Content.Shared.Mousetrap; // Starlight
```

CrazyPhantom779 commented on Resources/Locale/en-US/mousetraps/mousetraps.ftl:
```diff
@@ -1,2 +1,3 @@
 mousetrap-on-activate = The mousetrap was activated.
 mousetrap-on-deactivate = The mousetrap was deactivated.
+mousetrap-cat-avoid = deftly avoids the mousetrap
```
Move this to _starlight

CrazyPhantom779 commented on Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:
```diff
@@ -3267,7 +3268,6 @@
       - DoorBumpOpener
   - type: MovementAlwaysTouching
   - type: PressureImmunity
-  - type: ProtectedFromStepTriggers
   - type: Insulated
```
```suggestion
  - type: PressureImmunity
//  - type: ProtectedFromStepTriggers // Starlight edit
  - type: Insulated
```

Rinary1 replied:
```suggestion
  - type: PressureImmunity
#  - type: ProtectedFromStepTriggers Starlight-edit: Moved up to cat so they won't die from mousetraps.
  - type: Insulated
```

CrazyPhantom779 commented on Content.Server/Mousetrap/MousetrapSystem.cs:
```diff
@@ -0,0 +1,29 @@
+        _chat.TrySendInGameICMessage(args.Tripper, message, InGameICChatType.Emote, ChatTransmitRange.Normal, hideLog: true);
+    }
+    // Starlight-end
+}
```
Why are you making a new file but also including starlight comments in it? either move this to _starlight or move the changes to the other ``MousetrapSystem.cs``

Rinary1 reviewed (changes requested):
you have incorrect changelog message(author).

Rinary1 commented on Content.Server/Mousetrap/MousetrapSystem.cs:
```diff
@@ -0,0 +1,29 @@
+    private void OnMousetrapAvoided(Entity<MousetrapComponent> ent, ref MousetrapAvoidedEvent args)
+    {
+        var message = Loc.GetString("mousetrap-cat-avoid");
+        _chat.TrySendInGameICMessage(args.Tripper, message, InGameICChatType.Emote, ChatTransmitRange.Normal, hideLog: true);
```
this should be maked as popup, also you can just make this at the shared side instead of server so you don't need to create unnecessary system.

PGrayCS committed:
Address PR review feedback

- Changed emote to popup (SharedPopupSystem in shared code)
- Removed server-side MousetrapSystem.cs (no longer needed)
- Moved locale to _Starlight/mousetrap/mousetrap.ftl
- Comment out removed ProtectedFromStepTriggers instead of deleting
- Added // Starlight comments to imports

beck-thompson commented on Content.Shared/StepTrigger/Systems/StepTriggerImmuneSystem.cs:
```diff
@@ -20,6 +23,13 @@ private void OnStepTriggerClothingAttempt(Entity<PreventableStepTriggerComponent
+            // Starlight-start: show popup when entity avoids mousetrap
+            if (HasComp<MousetrapComponent>(ent))
+            {
+                _popup.PopupEntity(Loc.GetString("mousetrap-avoided", ("entity", args.Tripper)), ent);
```
just remove the hascomp who cares about the popup - why does the popup specifically care about mousetraps anyway? tell the ai to make a more generic solution or just tell it to remove it
